### PR TITLE
Refactoring Part 4 - SharedUtils.Misc

### PR DESCRIPTION
**Pull request by TracerDS**

### CODE REVIEW

File: Shared/sdk/SharedUtil.Misc.h [hunk 1]
```diff
@@ -604,24 +546,23 @@ namespace SharedUtil
     //
     // Note: IDs run from 1 to Capacity
     //
-    template <typename T, unsigned long INITIAL_MAX_STACK_SIZE>
+    template <typename T, std::uint32_t INITIAL_MAX_STACK_SIZE>
```
Comment (tederis) [inline Shared/sdk/SharedUtil.Misc.h]: Better to use size_t here

File: Shared/sdk/SharedUtil.Misc.h [hunk 2]
```diff
@@ -604,24 +546,23 @@ namespace SharedUtil
...
         }
 
-        unsigned long GetCapacity() const { return m_ulCapacity; }
-
-        unsigned long GetUnusedAmount() const { return m_Queue.size(); }
+        std::uint32_t GetCapacity() const noexcept { return m_ulCapacity; }
+        std::uint32_t GetUnusedAmount() const noexcept { return m_Queue.size(); }
 
-        void ExpandBy(unsigned long ulAmount)
+        void ExpandBy(std::uint32_t amount)
```
Comment (tederis) [inline Shared/sdk/SharedUtil.Misc.h]: size_t

Comment (tederis) [reply Shared/sdk/SharedUtil.Misc.h]: The same for all quantity types above and below.

File: Shared/sdk/SharedUtil.Misc.h (hunk 1 above)
Comment (Dutchman101) [reply Shared/sdk/SharedUtil.Misc.h]: @TracerDS

File: Shared/sdk/SharedUtil.Misc.h (hunk 2 above)
Comment (Dutchman101) [reply Shared/sdk/SharedUtil.Misc.h]: @TracerDS

**Comment by TracerDS** [comment]
Im not interested in advancing this PR further anymore.
If someone wants to refresh it, feel free to do so.

**Pull request closed without merging**